### ISSUE TITLE
Detect non-ASCII symbols in byte literals

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9244]
## Environment

* **IntelliJ Rust plugin version:** 0.4.177.4822-222-nightly
* **Rust toolchain version:** 1.65.0-nightly (20ffea693 2022-08-11) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.3.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
You cannot use non-ASCII symbols in byte literals. And currently, the plugin doesn't show the corresponding errors.
In addition, the plugin can suggest replacing such characters with sequences of the corresponding `\xHH` escapes

The corresponding compiler output:
```
error: non-ASCII character in byte constant
 --> src/main.rs:2:15
  |
2 |     let x = b'👋';
  |               ^^
  |               |
  |               byte constant must be ASCII
  |               this multibyte character does not fit into a single byte

error: non-ASCII character in byte constant
 --> src/main.rs:3:15
  |
3 |     let y = b"🔥🎉";
  |               ^^ byte constant must be ASCII
  |
help: if you meant to use the UTF-8 encoding of '🔥', use \xHH escapes
  |
3 |     let y = b"\xF0\x9F\x94\xA5🎉";
  |               ~~~~~~~~~~~~~~~~

error: non-ASCII character in byte constant
 --> src/main.rs:3:16
  |
3 |     let y = b"🔥🎉";
  |                 ^^ byte constant must be ASCII
  |
help: if you meant to use the UTF-8 encoding of '🎉', use \xHH escapes
  |
3 |     let y = b"🔥\xF0\x9F\x8E\x89";
  |                 ~~~~~~~~~~~~~~~~

```

## Steps to reproduce
```rust
fn main() {
    let x = b'👋';
    let y = b"🔥🎉";
}
```
